### PR TITLE
Selenium tests v3

### PR DESCRIPTION
**Pull request by Melduous**
Implements #294 

@shapiromatron Checked in local branch, and created PR.

### CODE REVIEW

**Comment by shapiromatron** [comment]
@Melduous I pushed a bunch more changes but github is acting weird right now...

**Comment by shapiromatron** [comment]
@Melduous do you mind taking a look to see if you can fix this last failing test?

**Comment by Melduous** [comment]
@shapiromatron Yep, taking a look now.

File: tests/frontend/integration/test_integrations.py [hunk 1]
```diff
@@ -35,3 +35,31 @@ def test_summary_visual_browse(self):
 
     def test_user_permissions(self):
         tests.user_permissions(self.driver, self.live_server_url)
+
+    # tests for /lit/assessment/2/references/ unworking
```
Comment (shapiromatron) [inline tests/frontend/integration/test_integrations.py:39]: not sure what this means - remove comment or describe issue?

File: tests/frontend/integration/tests/bioassay.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        if "2 year bioassay" in elem.text:
+            hasValueInTable = True
+            break
+
+    assert hasValueInTable
+
+    # confirm "tester" exists under Available animal groups
+    assert h.Text(to_left_of="rat").value == "tester"
+
+    # TODO: /animal-group/
```
Comment (shapiromatron) [inline tests/frontend/integration/tests/bioassay.py:28]: ok to do in this PR?

File: tests/frontend/integration/tests/visuals.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    assert len(driver.find_elements_by_css_selector("svg")) > 0
+    assert len(driver.find_elements_by_css_selector("svg .x_gridlines")) > 0
+    assert len(driver.find_elements_by_css_selector("svg .y_gridlines")) > 0
+
+    h.go_to(root_url + "/summary/visual/7/")
+
+    h.wait_until(h.Text("embedded-tableau").exists)
+
+    # ensure iframe exists inside .tableauPlaceholder -- this is causing problems
+    # assert len(driver.find_elements_by_tag_name("iframe")) > 0
```
Comment (shapiromatron) [inline tests/frontend/integration/tests/visuals.py:48]: anything we can do about this?

**Review by shapiromatron — CHANGES_REQUESTED** [review]
I saw a few more TODOs and things, can those be addressed in this PR or let's better document the TODO to indicate why they weren't done...

**Comment by Melduous** [comment]
Will look at these and get back to you shortly, honestly didn't look at anything other than the cleanup.py issue since circling back around here, so I believe most of these are solvable.

File: tests/frontend/integration/tests/visuals.py (hunk 3 above)
Comment (Melduous) [reply tests/frontend/integration/tests/visuals.py]: This has been fixed in the latest changeset

File: tests/frontend/integration/tests/bioassay.py (hunk 2 above)
Comment (Melduous) [reply tests/frontend/integration/tests/bioassay.py]: This has been fixed in the latest changeset.

File: tests/frontend/integration/test_integrations.py (hunk 1 above)
Comment (Melduous) [reply tests/frontend/integration/test_integrations.py:39]: I'm still thinking on this one, this seems like some sort of race condition(I think, even though I've just introduced a 5 second wait in one of my tests) with event binding and React that Selenium isn't handling well, I've tried clicking with Helium, Selenium, and injecting raw javascript into the browser. So I'm thinking through next steps here.

Comment (Melduous) [reply tests/frontend/integration/test_integrations.py:39]: This ended up being a simple typo, that I had apparently been overlooking the whole time, Nice.

**Comment by Melduous** [comment]
@shapiromatron I believe this is ready for your eyes when you have the time to give it another look.

**Review by shapiromatron — APPROVED** [review]

**Comment by shapiromatron** [comment]
@Melduous thank you, nice job!

**Pull request merged**